File: 6h15m/programmers/level1/210516-PEclothes.js
// 실패(아직 이유 못 찾음)

function solution(n, lost, reserve) {
    var temp = 0;
    lost.sort(function(a, b){return a-b;});
    reserve.sort(function(a, b){return a-b;});
    for(var i = 0; i < lost.length; i++){
        if(reserve.includes(lost[i])){
            reserve.splice(reserve.indexOf(lost[i]), 1);
            temp += 1;
        } else if(reserve.includes(lost[i] - 1)){
            if(lost.includes(lost[i] - 1)){
                continue;
            }
            reserve.splice(reserve.indexOf(lost[i] - 1), 1);
            temp += 1;
        } else if(reserve.includes(lost[i] + 1)){
            if(lost.includes(lost[i] + 1)){
                continue;
            }
            reserve.splice(reserve.indexOf(lost[i] + 1), 1);
            temp += 1;
        }
    return n - (lost.length - temp);
    }
}

File: 6h15m/programmers/python/level1/210516-PEclothes.py
#프로그래머스 | 체육복 (https://programmers.co.kr/learn/courses/30/lessons/42862)
def solution(n, lost, reserve):
    temp = 0
    lost.sort()
    reserve.sort()
    for l in lost:
        if l in reserve:
            reserve.remove(l)
            temp += 1
        elif l - 1 in reserve:
            if l - 1 in lost:
                continue
            reserve.remove(l - 1)
            temp += 1
        elif l + 1 in reserve:
            if l + 1 in lost:
                continue
            reserve.remove(l + 1)
            temp += 1
            
    return n - (len(lost) - temp)